tests/array.rs

```diff
@@ -1954,6 +1954,14 @@ fn test_array_clone_same_view() {
     assert_eq!(a, b);
 }
 
+#[test]
+fn test_array2_from_diag() {
```

Comment:
It would be good to also include a test for when `diag` is zero-length. (In my experience writing code for `ndarray`, it's common to have bugs related to zero-size arrays, so it's useful to have tests for those cases.)